packages/fxa-settings/src/lib/utilities.ts

```diff
@@ -129,3 +129,26 @@ export function once(key: string, callback: () => void) {
 export function resetOnce() {
   calls = new Set<string>();
 }
+
+/**
+ * Constructs a URL with UTM parameters appended to the query string.
+ *
+ * @param {string} url - The base URL.
+ * @param {string} utmMedium - The medium through which the link is being shared (e.g., mozilla-websites, product-partnership, email).
+ * @param {string} utmSource - The source of the traffic (e.g., moz-account).
+ * @param {string} utmTerm - The search term or keyword associated with the campaign.
+ * @param {string} utmContent - The specific content or product that the link is associated with (e.g., monitor, relay).
+ * @param {string} utmCampaign - The name of the marketing campaign (e.g., permanent, settings-promo).
+ * @returns {string} - The constructed URL with UTM parameters.
+ *
+ */
+export const constructHrefWithUtm = (
```

Comment:
A couple improvements definitely not required:
* Some of these options could get defaults
* Instead of 'string' type, we could list all the possibilities